text/src/swam/witx/WitxParser.scala

```diff
@@ -0,0 +1,53 @@
+package swam
+package witx
+
+import java.nio.file.Path
+
+import fastparse._
+import swam.text.{ParserException, readFile}
+import swam.witx.parser.{ImportContext, ModuleParser, TypesParser}
+import swam.witx.unresolved.{BaseWitxType, ModuleInterface}
+import cats.effect._
+import cats.implicits._
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-20
+  */
+class WitxParser[F[_]](implicit val F: Effect[F]) {
```

Comment:
You probably only need a `Sync` not an `Effect`. I fixed this on master, Swam use to required `Effect`s everywhere but it was a mistake. Keep the typeclass constraint to the lowest required.